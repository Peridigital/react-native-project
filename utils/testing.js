import ShallowRenderer from 'react-test-renderer/shallow'

/**
 * Creates a shallow rendered string representation of a given react component
 *
 * @param {any} component - A React component to be snapshotted ex: <Text>Hello World<Text/>
 *
 * @returns {string} A shallow rendered string representation of the React component
 */
export function shallow (component) {
  const renderer = new ShallowRenderer()
  renderer.render(component)
  return renderer.getRenderOutput()
}
