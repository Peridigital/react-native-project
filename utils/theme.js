const colors = {
  primaryBackground: '#0C0D12',
  secondaryBackground: '#151821',
  primaryText: '#EAEAEA',
  border: '#7E4033',
  borderHighlight: '#D0A499'
}

export default colors
