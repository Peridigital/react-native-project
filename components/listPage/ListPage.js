import React from 'react'
import { noop } from 'lodash'
import {
  View,
  Text,
  Pressable,
  FlatList
} from 'react-native'
import PropTypes from 'prop-types'
import style from './ListPage.style'
import { generateListingComponent } from './ListPage.utils'

export function ListPage ({ starships, navigateToDetails }) {
  return (
    <View style={style.container}>
      <Text> Hi there</Text>
      <Pressable
        onPress={navigateToDetails}
        style={style.button}
      ><Text>Click here</Text></Pressable>
      <FlatList
          data={starships}
          renderItem={generateListingComponent(navigateToDetails)}
        />
    </View>
  )
}

ListPage.propTypes = {
  starships: PropTypes.array,
  navigateToDetails: PropTypes.func
}

ListPage.defaultProps = {
  navigateToDetails: noop
}

export default ListPage
