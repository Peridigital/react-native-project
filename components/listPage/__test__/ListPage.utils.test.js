/* eslint-disable no-undef */
import { generateNavigateToDetailsWithId, generateListingComponent } from '../ListPage.utils'

describe('generateNavigateToDetailsWithId', () => {
  it('should toggle from true to false', () => {
    const navigation = {
      navigate: jest.fn()
    }
    const id = 1
    generateNavigateToDetailsWithId({ navigation })(id)()

    expect(navigation.navigate).toHaveBeenCalledWith({
      name: 'Details',
      params: { id }
    })
  })
})

describe('generateListingComponent', () => {
  it('should toggle from true to false', () => {
    const navigateToDetails = jest.fn()
    const wrapper = generateListingComponent(navigateToDetails)({ item: { name: 'Big Ship', model: 'Big Ship Model' } })

    expect(wrapper).toMatchSnapshot()
  })
})
