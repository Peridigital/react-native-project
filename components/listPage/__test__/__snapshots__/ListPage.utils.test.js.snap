// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateListingComponent should toggle from true to false 1`] = `
<Listing
  hasFontsLoaded={false}
  navigateToDetails={[Function]}
  starship={
    {
      "model": "Big Ship Model",
      "name": "Big Ship",
    }
  }
/>
`;
